### PR TITLE
CAM-10646: Prevent NullPointerException during destruction if delegate isn't initialized.

### PR DESCRIPTION
**Pull request by tobiasschaefer**

### CODE REVIEW

**Comment by tobiasschaefer** [comment]
Please note that the build failure isn't related to my changes:

`Could not resolve dependencies for project org.camunda.bpm.springboot.project:qa-webapp-ee-runtime:jar:3.4.0-SNAPSHOT: Failure to find org.camunda.bpm.webapp:camunda-webapp-webjar-ee:jar:7.12.0-SNAPSHOT in https://oss.sonatype.org/content/repositories/snapshots/ was cached in the local repository, resolution will not be reattempted until the update interval of sonatype-snapshots has elapsed or updates are forced`

**Comment by yanavasileva** [comment]
Hi @tobiasschaefer ,

Thank you for your contribution.
We are aware of the failing travis build, please ignore it for now.
I created [CAM-10646](https://app.camunda.com/jira/browse/CAM-10646) where you can track the progress of PR.

Best regards,
Yana

**Comment by yanavasileva** [comment]
Hi @tobiasschaefer,

Could you please share with us what was your use case where you stumbled upon this error?

Best regards,
Yana

**Comment by tobiasschaefer** [comment]
Hi @yanavasileva,

I currently don't have the time to give you an exact reproduction scenario but it basically happens if the datasource is misconfigured. I then had the NullPointerException which I fixed in the pull request instead of seeing the stacktrace of the root cause of the data source misconfiguration.

Kind Regards
Tobias

File: starter-webapp-core/src/test/java/org/camunda/bpm/spring/boot/starter/webapp/filter/LazyDelegateFilterTest.java [hunk 1]
```diff
@@ -17,12 +17,7 @@
 package org.camunda.bpm.spring.boot.starter.webapp.filter;
 
 import static org.junit.Assert.assertSame;
-import static org.mockito.Mockito.doReturn;
-import static org.mockito.Mockito.inOrder;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.spy;
-import static org.mockito.Mockito.times;
-import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.*;
```
Comment (koevskinikola) [inline starter-webapp-core/src/test/java/org/camunda/bpm/spring/boot/starter/webapp/filter/LazyDelegateFilterTest.java]: Hi @tobiasschaefer, 

According to the project guidelines, we don't allow wildcard imports. So I would restore the separate imports and just add the `org.mockito.Mockito.never` static import:
```suggestion
import static org.mockito.Mockito.doReturn;
import static org.mockito.Mockito.inOrder;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.never;
import static org.mockito.Mockito.spy;
import static org.mockito.Mockito.times;
import static org.mockito.Mockito.verify;
```

Best,
Nikola

**Comment by tobiasschaefer** [comment]
Hi @koevskinikola ,

I updated the pull request as requested.

Please note: if you do a full text search for `.*;` you will find three other existing places where wildcard imports are being used. So I thought it would be fine.

Kind Regards
Tobias

**Comment by koevskinikola** [comment]
Hi Tobias,

Thanks! Thank you for telling me. We try to make the code as consistent as possible, but sometimes things slip through.

I'll merge the PR with master. Thanks again for the contribution!

Best,
Nikola

**Pull request merged**